packages/babel-plugin/src/class-names/index.tsx

```diff
@@ -13,4 +63,55 @@ import { State } from '../types';
  * @param path Babel path - expects to be a JSX opening element.
  * @param state Babel state - should house options and meta data used during the transformation.
  */
-export const visitClassNamesPath = (_: NodePath<t.JSXOpeningElement>, __: State) => {};
+export const visitClassNamesPath = (path: NodePath<t.JSXElement>, meta: Metadata) => {
+  if (
+    t.isJSXIdentifier(path.node.openingElement.name) &&
+    path.node.openingElement.name.name !== meta.state.compiledImports?.ClassNames
+  ) {
+    // We aren't interested in this element. Bail out!
+    return;
+  }
+
+  const children = getJsxChildrenAsFunction(path);
```

Comment:
Why we extracted children so early in the flow? 🤔 Can't we move it closer to its usage before line 155?